tpcc: set isolation level to serializable in txns

rjnn opened the pull request:
This commit adds support for explicitly setting the isolation level. Doing this work revealed that Cockroach doesn't actually support this syntax (although this syntax really is to ensure serializability in Postgres so that we are comparing apples to apples), so this isn't ready for merging until that is fixed.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/loadgen/69)
<!-- Reviewable:end -->

andreimatei commented:
which syntax don't we support? `SET TRANSACTION ISOLATION LEVEL` is supported.

---

Review status: 0 of 5 files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/loadgen/69#-:-Krh94liRAYFAkRuUefh:buxdlnl)*
<!-- Sent from Reviewable.io -->

jordanlewis commented:
I believe neither the READ nor WRITE keywords are supported there.

rjnn commented:
https://github.com/cockroachdb/cockroach/issues/17709
On Wed, Aug 16, 2017 at 6:27 PM Andrei Matei <notifications@github.com>
wrote:

> which syntax don't we support? SET TRANSACTION ISOLATION LEVEL is
> supported.
> ------------------------------
>
> Review status: 0 of 5 files reviewed at latest revision, all discussions
> resolved, some commit checks failed.
> ------------------------------
>
> *Comments from Reviewable
> <https://reviewable.io:443/reviews/cockroachdb/loadgen/69#-:-Krh94liRAYFAkRuUefh:buxdlnl>*
>
> —
> You are receiving this because you authored the thread.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/cockroachdb/loadgen/pull/69#issuecomment-322917037>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABfkifUXS550jUDMVOiG8LlE-JHNsfrDks5sY2zggaJpZM4O5hpR>
> .
>

Commit:
tpcc: force serializable isolation

Postgres by default provides snapshot isolation, whereas CockroachDB
provides serializability, which makes apples to apples comparisons
potentially misleading. Force serializability explicitly.
